src/sentry/integrations/gitlab/client.py

```diff
@@ -115,23 +115,44 @@ def _refresh_auth(self):
 
     def request(self, *args: Any, **kwargs: Any):
         if SiloMode.get_current_mode() == SiloMode.REGION:
+            # Skip token refreshes in Region silo, as these will
+            # be handled below by the control silo when the
+            # integration proxy invokes the client code.
             return super().request(*args, **kwargs)
-        # Only perform the refresh token flow in either monolithic or the control silo mode.
+
+        return self._issue_request_with_auto_token_refresh(*args, **kwargs)
+
+    def _issue_request_with_auto_token_refresh(self, *args: Any, **kwargs: Any):
         try:
-            return super().request(*args, **kwargs)
+            response = super().request(*args, **kwargs)
         except ApiUnauthorized as e:
             if self.is_refreshing_token:
                 raise e
+            return self._attempt_request_after_refreshing_token(*args, **kwargs)
```

Comment:
This should only ever be hit in monolith mode. The integration proxy passes `raw_response: True` when it issues requests, meaning we don't ever construct or raise Api Exceptions.